src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -137,87 +140,90 @@ class DuplexConsensusCaller(override val readNamePrefix: String,
-          val ss2Flag = singleStrand2.head.negativeStrand
-          if (singleStrand2.exists(_.negativeStrand != ss2Flag)) {
-            val ss2MI   = singleStrand2.head.apply[String](ConsensusTags.MolecularId)
-            rejectRecords(ab ++ ba, FilterCollision)
-            logger.debug(s"Not all AB-R2s and BA-R1s were on the same strand for molecule with id: $ss2MI")
-            return Nil
-          }
-        }
+    // Partition reads into those that end with /A and those that end with /B (verify the latter
+    val (ab, ba) = pairs.partition(r => r[String](ConsensusTags.MolecularId).endsWith("/A"))
 
-        // Filter by common indel pattern with AB and BA together
-        val filteredXs = filterToMostCommonAlignment((abR1s ++ baR2s).flatMap(toSourceRead(_, this.minInputBaseQuality, this.trim)))
-        val filteredYs = filterToMostCommonAlignment((abR2s ++ baR1s).flatMap(toSourceRead(_, this.minInputBaseQuality, this.trim)))
-
-        // Then split then back apart for SS calling
-        val filteredAbR1s = filteredXs.filter(_.sam.exists(_.firstOfPair))
-        val filteredAbR2s = filteredYs.filter(_.sam.exists(_.secondOfPair))
-        val filteredBaR1s = filteredYs.filter(_.sam.exists(_.firstOfPair))
-        val filteredBaR2s = filteredXs.filter(_.sam.exists(_.secondOfPair))
-
-        // Call the single-stranded consensus reads
-        val abR1Consensus = ssCaller.consensusCall(filteredAbR1s)
-        val abR2Consensus = ssCaller.consensusCall(filteredAbR2s)
-        val baR1Consensus = ssCaller.consensusCall(filteredBaR1s)
-        val baR2Consensus = ssCaller.consensusCall(filteredBaR2s)
-
-        // Call the duplex reads
-        val duplexR1Sources = filteredAbR1s ++ filteredBaR2s
-        val duplexR2Sources = filteredAbR2s ++ filteredBaR1s
-        val duplexR1 = duplexConsensus(abR1Consensus, baR2Consensus, duplexR1Sources)
-        val duplexR2 = duplexConsensus(abR2Consensus, baR1Consensus, duplexR2Sources)
-
-        // Convert to SamRecords and return
-        (duplexR1, duplexR2) match {
-          case (Some(r1), Some(r2)) =>
-            Seq(
-              createSamRecord(r1, FirstOfPair, toUmiBases(duplexR1Sources)),
-              createSamRecord(r2, SecondOfPair, toUmiBases(duplexR2Sources))
-            )
-          case _                    =>
-            // NB: some reads may have been rejected already in filterToMostCommonAlignment, so just
-            //     reject those records that survived the initial filtering.
-            val remainingRecs = filteredXs ++ filteredYs
-            rejectRecords(remainingRecs.flatMap(_.sam), FilterSsConsensus)
-            Nil
+    // Verify ba reads end with /B
+    require(ba.forall(r =>  r[String](ConsensusTags.MolecularId).endsWith("/B")), "Expected MIs to end with /A or /B")
+    // Verify they all have the same molecular identifiers
+    require(pairs.map(_.apply[String](ConsensusTags.MolecularId).dropRight(2)).distinct.length <= 1,
```

Comment:
This would break too if we change GroupReadsByUmi.  Please use `sourceMoleculeId(rec)`.

Reply:
Also, this is already enforced by the iterator machinery that calls into this, I'm not sure we need to re-validate here.